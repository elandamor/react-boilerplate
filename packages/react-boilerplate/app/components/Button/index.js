/**
 * Button
 * @type {Component} Presentational
 */

import React from 'react';
import PropTypes from 'prop-types';
// Styled-Components
import Wrapper from './styles';

const Button = ({ children }) => (
  <Wrapper>{children}</Wrapper>
);

Button.propTypes = {
  children: PropTypes.any.isRequired,
};

export default Button;
