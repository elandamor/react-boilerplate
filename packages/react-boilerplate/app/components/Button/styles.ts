import styled from 'styled-components';

const Wrapper = styled.button`

`;

export default Wrapper;

export {

};
