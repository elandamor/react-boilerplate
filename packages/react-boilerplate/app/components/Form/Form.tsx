import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name Form component
 * @description Form component.
 * @example
 * <Form />
 */

interface IProps {
  children?: JSX.Element[];
  className?: string;
  onSubmit: () => any;
}

const Form: SFC<IProps> = ({ className, children, onSubmit }) => (
  <Wrapper
    className={classNames('c-form', className)}
    onKeyDown={
      (event) => {
        /**
         * Note: Pressing enter in some input in a browser forms
         *  triggers onClick on the first child button
         *
         * So, prevent `enter` from triggering `onClick` on any buttons
         *  and instead trigger onSubmit
         */
        if (event.key === 'Enter') {
          event.preventDefault();
          onSubmit();
        }
      }
    }
    onSubmit={
      (event) => {
        /**
         * Prevent submit from reloading the page
         */
        event.preventDefault();
        event.stopPropagation();
        onSubmit();
      }
    }
  >
    {children}
  </Wrapper>
);

Form.defaultProps = {
  className: '',
};

export default Form;
