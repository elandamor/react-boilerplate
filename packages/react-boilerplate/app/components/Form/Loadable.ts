/**
 *
 * Asynchronously loads the component for Form
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
