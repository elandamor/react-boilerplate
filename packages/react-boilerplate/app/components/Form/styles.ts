import styled from 'styled-components';

const Wrapper = styled.form`

`;

export default Wrapper;

export {

};
