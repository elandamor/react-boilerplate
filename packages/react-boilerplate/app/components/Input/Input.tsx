import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name Input component
 * @description Input component.
 * @example
 * <Input
 *  id="text"
 *  label="Text"
 *  type="text"
 * />
 */

interface IProps {
  className?: string;
  id: string;
  label: string;
  sronly?: boolean;
  type?: string;
}

const Input: SFC<IProps> = ({ className, id, label, sronly, ...rest }) => (
  <Wrapper className={classNames('c-input-wrapper', className)}>
    <label htmlFor={id}>
      <span
        className={classNames('a-label', {
          'sr-only': sronly,
        })}
      >
        { label }
      </span>
      <input id={id} className="a-input" {...rest} />
    </label>
  </Wrapper>
);

Input.defaultProps = {
  sronly: false,
  type: 'text',
};

export default Input;
