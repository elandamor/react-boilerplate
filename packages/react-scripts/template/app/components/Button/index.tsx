import React, { MouseEvent, FC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

export interface IButtonProps {
  backgroundColor?: string;
  className?: string;
  disabled?: boolean;
  icon?: JSX.Element;
  iconOnly?: boolean;
  iconPosition?: string;
  iconSize?: number;
  ml?: number | string;
  mr?: number | string;
  mx?: number | string;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  outlined?: boolean;
  raised?: boolean;
  text?: string;
  textColor?: string;
  type?: 'button' | 'submit';
}

/**
 * @render react
 * @name Button component
 * @description Defines a clickable button.
 * @example
 * <Button text="Test" />
 */

const Button: FC<IButtonProps> = ({
  className,
  onClick: handleClick,
  text,
  ...rest
}) => (
  <Wrapper
    className={classNames('c-btn', className)}
    onClick={handleClick}
    {...rest}
  >
    {rest.icon && rest.iconPosition === 'left' && <i>{rest.icon}</i>}
    {!rest.iconOnly && <label>{text}</label>}
    {rest.icon && rest.iconPosition === 'right' && <i>{rest.icon}</i>}
  </Wrapper>
);

Button.defaultProps = {
  iconOnly: false,
  iconPosition: 'left',
  onClick: () => null,
  text: 'Button',
  type: 'button',
};

export default Button;
