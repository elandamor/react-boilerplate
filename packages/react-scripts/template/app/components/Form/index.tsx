import React, { FC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

// import { makeDebugger } from '../../utils';
// const debug = makeDebugger('Form');

interface IProps {
  children: JSX.Element | JSX.Element[];
  className?: string;
  onSubmit?: (event?: React.FormEvent<HTMLFormElement>) => void;
}

/**
 * @render react
 * @name Form component
 * @description Creates an HTML form for user input.
 * @example
 *  <Form onSubmit={() => handleSubmit}>
 *    <Input
 *      id="text"
 *      label="Text"
 *      name="text"
 *      type="text"
 *    />
 *    <Button
 *      text="Submit"
 *      type="submit"
 *    />
 *  </Form>
 */

const Form: FC<IProps> = ({ children, className, onSubmit: handleSubmit }) => (
  <Wrapper
    children={children}
    className={classNames('c-form', className)}
    onKeyDown={(event: React.KeyboardEvent<HTMLFormElement>) => {
      /**
       * Note: Pressing enter in some input in a browser forms triggers onClick
       * on the first child button
       *
       * So, prevent `enter` from triggering `onClick` on any buttons and
       * instead trigger onSubmit
       */
      if (event.key === 'Enter') {
        // @ts-ignore
        if (event.target.type !== 'textarea') {
          event.preventDefault();
        }
      }
    }}
    onSubmit={(event: React.FormEvent<HTMLFormElement>) => {
      /**
       * Prevent submission from reloading the page
       */
      event.preventDefault();
      event.stopPropagation();
      handleSubmit && handleSubmit(event);
    }}
  />
);

Form.defaultProps = {
  onSubmit: () => null,
};

export default Form;
