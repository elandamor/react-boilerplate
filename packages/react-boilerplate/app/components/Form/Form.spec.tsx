// Form.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Form from './index';

import Button from '../Button';
import Input from '../Input';

describe('<Form />', () => {
  it('should display a Form', () => {
    const component = renderer
      .create(
        <Form onSubmit={() => null}>
          <Input
            id="username"
            label="Username"
            name="username"
            onChange={() => null}
          />
          <Button onClick={() => null}>Submit</Button>
        </Form>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
