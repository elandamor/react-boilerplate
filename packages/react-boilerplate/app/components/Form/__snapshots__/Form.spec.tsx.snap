// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should display a Form 1`] = `
<form
  className="c-form "
  onKeyDown={[Function]}
  onSubmit={[Function]}
>
  <div
    className="c-input-wrapper "
  >
    <label
      htmlFor="username"
    >
      <span
        className="a-label"
      >
        Username
      </span>
      <input
        className="a-input"
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <button
    className="c-btn "
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
